### DISCUSSION TITLE
关于 COT prompt 的讨论

### DISCUSSION THREAD

**Discussion by goactiongo** [labring/FastGPT#2680] [Announcements]
4.8.9
知识库+ai的场景下

同样的两段提示词，
一个提示词是让模型输出思考过程，然后回复答案。
另一个提示词是让模型不要输出思考过程，直接回复答案。

两种方式回答的效果不一样，输出思考过程的回答ok，不输出思考过程的回答不好。

例如知识库中有很多电话
用户输入问题：电话

提示词要求【输出思考过程】时，ai回复有多个不同电话，需要进一步明确。。。。——这个效果还可以。

提示词要求【不输出思考过程】时，ai回复，没有找到结果。或者只回复了其中一个电话。——这个效果就不好

请大神帮忙测试下是否同样的问题，我更换了阿里和百度不同的模型效果几乎一样，例如阿里qwen72b   百度4.0turbo  智谱最新的模型。

这应该不是fastgpt的问题，但是很奇怪为什么不同厂家不同模型都是类似问题。

这样会导致类似用户提问“电话”这种模糊的场景下回复效果很差，明明知识库中有很多电话可以回复让用户进一步明确，但模型要么只回复其中一个电话，要么回复没有答案

**Comment by lijiajun1997** [comment]
这已经是共识，可以去看大模型思维链 CoT（Chain of Thought）相关文章

**Comment by c121914yu** [comment]
毕竟是基于概率的模型，如果入口的范围就很大，出来的结果大概率不好，必须一步步缩小可选概率范围。
